configure.ac

```diff
@@ -303,6 +303,16 @@ AC_CANONICAL_BUILD
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
 
+# Check for the --with-android option
+AC_ARG_WITH([android],
+  [AS_HELP_STRING([--with-android], [enable Android support])],
+  [android=yes], [android=no])
+
+# If --with-android is specified, add -landroid-shmem to oc_ldflags
+# This in shea Allah ensures that it compiles within termux/android.
+AS_IF([test "$with_android" = "yes"],
+    [oc_ldflags="$oc_flags -landroid-shmem"])
```

Comment:
The `autoconf` script should detect when the library is needed and add it automatically. Supposedly there is some way to detect that we are compiling in an Android system?